docs/rules/no-unreadable-array-destructuring.md

```diff
@@ -0,0 +1,20 @@
+# Disallow unreadable array destructuring
+
+Destructuring is very useful, but it can also make some code harder to read.
```

Comment:
It should describe what it defines as "unreadable", meaning, what it limits. I know we show that in the examples below, but doesn't hurt to describe it.

Reply:
I think we also should mention that you might have to modify the built-in `prefer-destructuring` rule to be compatible with this one. And how to modify it. Context: https://github.com/sindresorhus/eslint-plugin-unicorn/issues/197#issuecomment-427459976